### ISSUE TITLE
Help Wanted: Backtrace not reporting C++ crashes on our app.

### ISSUE THREAD

**Issue by andreivictor-gumi** [backtrace-labs/backtrace-android#36]
Our app is mostly C++ in a .so file. After putting all integration code, backtrace seem to respond on sending reports invoked manually through the API and for Java unhandled exceptions etc. But for NDK C++ crashes there is totally no report sent as the app just silently crashes.

My C++ test cases:
TEST CASE A: using
`__builtin_trap()`

TEST CASE B:
```
// Cpp null pointer crash
class Foo { public: void Bar() {} };
((Foo *) nullptr)->Bar();
```

Using the Backtrace example project the observed behaviors:
TEST CASE A --> crashes and see a report in dashboard
TEST CASE B --> doesn't crash app but sees a report in the dashboard

in both cases, the call stack is only from dalvik and java, but no Cpp callstack.

Now for our app,
TEST CASE A --> crash, no report in dashboard
TEST CASE B --> crash, no report in dashboard

So, how come there are no reports in both test cases? Is there something we missed or some underlying requirement to have C++ crash reporting work?

Here's my backtrace initialization in Java for reference.:
```
        private static void initializeBacktrace( boolean isResuming )
	{
		// Prepare backtrace credentials
		BacktraceCredentials credentials = new BacktraceCredentials( BuildConfig.BACKTRACE_UNIVERSE_ENDPOINT
				, BuildConfig.BACKTRACE_SUBMISSION_TOKEN );

		// Custom attributes to send over
		Map<String, Object> attributes = getDefaultAppAttributesForBacktrace();

		// Prepare Database
		Context context = getAppContext();
		String dbPath = getBacktraceDocumentDirectory() + "backtrace/"; // any path, eg. absolute path to the internal storage
		BacktraceDatabaseSettings settings = new BacktraceDatabaseSettings( dbPath );
		settings.setMaxRecordCount( 100 );
		settings.setMaxDatabaseSize( 1000 );
		settings.setRetryBehavior( RetryBehavior.ByInterval );
		settings.setAutoSendMode( true );
		settings.setRetryOrder( RetryOrder.Queue );
		BacktraceDatabase database = new BacktraceDatabase( context, settings );

		// Creating the actual Backtrace Client
		backtraceClient = new BacktraceClient( context, credentials, database, attributes, getDefaultAttachments() );
		LapisEnvConst.ENABLED_BACKTRACE = isBacktraceEnabled();

		// Note: After instantiation of BacktraceClient, all files inside the database path will be
		// cleared or deleted.

		if( !isResuming )
		{
			// Delete backtrace log file, if it exists
			String fullFilePath = getBacktraceLogFullPath();
			File file = new File( fullFilePath );
			if( file.exists() )
			{
				try
				{
					file.delete();
					Log.d( TAG, "Deleted file:" + fullFilePath );
				}
				catch( Exception e )
				{
					Log.e( TAG, e.getLocalizedMessage() );
				}
			}
		}

        // Enable metrics (must be BEFORE enableNativeIntegration)
        // Not yet supported, only Unity for now.
//        BacktraceMetricsSettings metricsSettings = new BacktraceMetricsSettings( credentials );
//        backtraceClient.metrics.enable(metricsSettings);

		// Enabling breadcrumbs
		//backtraceClient.enableBreadcrumbs( getAppContext() );

		// For Proguard
		//backtraceClient.enableProguard();

		// For capturing unhandled application exceptions
		BacktraceExceptionHandler.enable( backtraceClient );
		///BacktraceExceptionHandler.setCustomAttributes( attributes );

		// For capturing NDK crashes
		database.setupNativeIntegration( backtraceClient, credentials, true );

		// Enable Application Not Responding detection
		backtraceClient.enableAnr( BuildConfig.BACKTRACE_ANR_TIMEOUT );
	}
```

As you see, I've commented out other features just to simplify and match the call order of the example project.

**Comment by andreivictor-gumi** [comment]
Thanks for looking into this @rqbacktrace, in response to your questions:

1. Yes, `getBacktraceDocumentDirectory()` ensures a `/` at the end, so there is no issue for the correctness of path.
2. Yes, I was able to check and see if our app is creating the crashpad similar to your screenshot. Here is mine:
![image](https://user-images.githubusercontent.com/52743216/139618624-595c3a70-0cbe-4bf2-878a-4cb54a91680c.png)
3. Re: Test Case B -- Thanks for letting me know of the behaviour on your side. I was wondering why some situations this crashes and sometimes it does not. The reason this is my test case, is because this is the most frequent type of error our app has.
4. Re: Other crash reporters -- Yes we have. Another crash reporter that we are finding a replacement for is still integrated.  Let me try to remove/disable it and I will update this ticket of the result.

**Comment by andreivictor-gumi** [comment]
Hello @rqbacktrace ! Great news! Everything works as expected and nicely when I disabled the other crash reporter!

Thanks for your help and insight on this, I truly appreciate it!

Closes #36
